src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/SpatialCoordinateSystemParticipant.cs

```diff
@@ -163,12 +163,12 @@ private void SendCoordinateStateMessage()
 
                 message.Write(position);
                 message.Write(rotation);
+                message.Flush();
 
                 byte[] newCoordinateStatusMessage = stream.ToArray();
```

Comment:
You probably don't need to allocate an array to perform the checks below.

Reply:
the allocation seems more important for caching the information which it looks like we were failing to do here.